src/models.jl

```diff
@@ -1,6 +1,74 @@
 _maybe_fill(x, n) = x
 _maybe_fill(x::Number, n) = fill(x, n)
 
+function tight_binding(g::AbstractGraph; t=1.0, tp=0.0, h::Union{<:Real,Vector{<:Real}}=0)
+  h = _maybe_fill(h, nv(g))
+  ℋ = OpSum()
+  if !iszero(t)
+    for e in edges(g)
+      ℋ += -t, "Cdag", maybe_only(src(e)), "C", maybe_only(dst(e))
+      ℋ += -t, "Cdag", maybe_only(dst(e)), "C", maybe_only(src(e))
+    end
+  end
+  if !iszero(t')
+    # TODO, more clever way of looping over next to nearest neighbors?
```

Comment:
There should be some functions from `Graphs.jl` we can use for this, for example [neighborhood](https://juliagraphs.org/Graphs.jl/dev/algorithms/connectivity/#Graphs.neighborhood-Union{Tuple{T},%20Tuple{U},%20Tuple{AbstractGraph{T},%20Integer,%20Any},%20Tuple{AbstractGraph{T},%20Integer,%20Any,%20AbstractMatrix{U}}}%20where%20{U%3C:Real,%20T%3C:Integer}), i.e. it could be `setdiff(neighborhood(g, v, 2), neighbors(g, v))`.